File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int l = nums.size();
        if(l<3){
            return false;
        }
        int fo = INT_MAX;
        int so = INT_MAX;
        
        for(int i=0;i<l;i++){
            if(nums[i] > so){
                return true;
            }
            else if(nums[i] > fo && nums[i] < so){
                so = nums[i];
            }
            else if(nums[i] < fo){
                fo = nums[i];
            }
        }
        return false;
    }
};

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.java
class Solution {
    //Learnt from techdose and editorial
    public boolean increasingTriplet(int[] nums) {
        int firstVariable = Integer.MAX_VALUE;
        int secondVariable = Integer.MAX_VALUE;
        
        for(int i = 0; i<nums.length; i++){
            if(nums[i] <= firstVariable){
                firstVariable = nums[i];
            }
            else if(nums[i] <= secondVariable){
                secondVariable = nums[i];
            }
            else{
                return true;
            }
        }
        return false;
    }
}